compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -243,6 +244,7 @@ impl<'psess, 'src> StringReader<'psess, 'src> {
                     let prefix_span = self.mk_sp(start, lit_start);
                     return (Token::new(self.ident(start), prefix_span), preceded_by_whitespace);
                 }
+                rustc_lexer::TokenKind::GuardedStrMaybe => self.report_guarded_str(start, str_before),
```

Comment:
```suggestion
                rustc_lexer::TokenKind::GuardedStrMaybe => self.maybe_report_guarded_str(start, str_before),
```
We do not always report a guarded str error/lint here.